src/client/WebClient.ts

```diff
@@ -266,6 +269,27 @@ export class WebClient extends Client {
 		}
 	}
 
+	async openLivechatRoom(rid: string): Promise<void> {
+		const end = prom.openRoom.startTimer();
+		const endAction = prom.actions.startTimer({ action: 'openRoom' });
+		try {
+			const calls: Promise<unknown>[] = [
+				this.client.methodCall('loadHistory', rid, null, 50, new Date()),
+			];
+
+			calls.push(this.client.methodCall('getRoomRoles', rid));
+
+			await Promise.all(calls);
```

Comment:
I was expecting to have `getVisitorInfo` among this `calls` .. they always happen together.. the way this was designed was to include everything that happens when the "real user" does this action.

in this scenario, is there any other calls that should be added here?

Reply:
We can add it here. I separated the visitor info as its own event because it happens on multiple places of the UI, like when you click on the forward chats, the left panel, the user badge, etc. So, since it happened multiple times, i tried to make it with its own frequency 👀 

(I will add a mandatory call here tho, since when you open the room, it will for sure call the visitor info)